src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/Path.java

```diff
@@ -123,8 +123,26 @@ public boolean pathsAreTheSame(final Path<T,E> path) {
     public boolean containsVertex(final T v) {
         Utils.nonNull(v, "Vertex cannot be null");
 
-        // TODO -- warning this is expensive.  Need to do vertex caching
-        return getVertices().contains(v);
+
+        if ( getEdges().isEmpty() ) {
+            return false;
+        } else {
+            boolean first = true;
+            for ( final E e : getEdges() ) {
```

Comment:
could you change the return type of `getEdges()` to `List`?  It's bothering me that its return type is `Collection`